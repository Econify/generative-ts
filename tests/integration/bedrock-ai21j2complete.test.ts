import {
  Ai21Jurassic2Api,
  createAwsBedrockModelProvider,
} from "@packages/core";

test("Bedrock - AI21 J2 Complete", async () => {
  // arrange
  const j2 = createAwsBedrockModelProvider({
    api: Ai21Jurassic2Api,
    modelId: "ai21.j2-mid-v1",
  });

  // act
  const response = await j2.sendRequest({
    prompt: "Brief history of NY Mets:",
    numResults: 1,
    maxTokens: 50,
    minTokens: 0,
    temperature: 0.5,
    topP: 1,
    topKReturn: 1,
    stopSequences: ["."],
  });

  // assert
  expect(response).toMatchApiSnapshot();
});
