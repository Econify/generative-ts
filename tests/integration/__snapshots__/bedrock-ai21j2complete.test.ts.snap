// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bedrock - AI21 J2 Complete 1`] = `
{
  "completions": [
    {
      "data": {
        "text": "string",
        "tokens": [
          {
            "generatedToken": {
              "logprob": "number",
              "raw_logprob": "number",
              "token": "string",
            },
            "textRange": {
              "end": "number",
              "start": "number",
            },
            "topTokens": [
              {
                "logprob": "number",
                "raw_logprob": "number",
                "token": "string",
              },
            ],
          },
        ],
      },
      "finishReason": {
        "reason": "string",
        "sequence": "string",
      },
    },
  ],
  "id": "number",
  "prompt": {
    "text": "string",
    "tokens": [
      {
        "generatedToken": {
          "logprob": "number",
          "raw_logprob": "number",
          "token": "string",
        },
        "textRange": {
          "end": "number",
          "start": "number",
        },
        "topTokens": [
          {
            "logprob": "number",
            "raw_logprob": "number",
            "token": "string",
          },
        ],
      },
    ],
  },
}
`;
